src/gcode-machine-control_test.cc

```diff
@@ -219,6 +227,25 @@ TEST(GCodeMachineControlTest, straight_segments_speed_change) {
   harness.gcode_emit()->motors_enable(false);  // finish movement.
 }
 
+TEST(GCodeMachineControlTest, m181_stops_and_set_planner_queue) {
+  static const struct LinearSegmentSteps expected[] = {
+    {0.0, 0.0, END_SENTINEL, {}},
+  };
+  Harness harness(expected);
+  GCodeParser::Config config;
+  GCodeParser::Config::ParamMap parameters;
+  config.parameters = &parameters;
+  GCodeParser parser = GCodeParser(config, harness.gcode_emit());
+  EXPECT_EQ(0, parser.error_count());
+
+  harness.gcode_emit()->gcode_start(&parser);
+  parser.ParseBlock("M181 S100", nullptr);
```

Comment:
Also add tests for `M181` without setting the S value (as it then should go back to planner size) and with negative values as well as too large values to see how things behave.

Reply:
(solved)